### ISSUE TITLE
[CrunchyrollBeta] Extractor KeyError

### ISSUE THREAD

**Issue by krandor** [yt-dlp/yt-dlp#3158]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.03.08.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

United States

### Description

Getting an extractor error for CrunchyrollBeta. I tried to do some debugging and it looks like the `window.__INITIAL_STATE__` json doesn't have any info. I'm assuming the `byId` element should have data in it since line 719 in the `crunchyoll.py` extractor says this: `episode_data = initial_state['content']['byId'][internal_id]`.

```
 "content": {
    "byId": {},
    "media": {
      "byId": {}
    },
    "playheads": {
      "byId": {}
    },
    "ratings": {
      "starRating": {
        "allIds": [],
        "byId": {},
        "userRatingById": {}
      },
      "thumbRating": {
        "allIds": [],
        "byId": {},
        "userRatingById": {}
      }
    },
    "reviews": {
      "activeContentId": null,
      "byId": {},
      "allIds": [],
      "isLoading": false,
      "total": 0,
      "error": null,
      "currentSortValue": "helpful",
      "currentFilterValue": "",
      "ownReview": null,
      "isCreateReviewLoading": false,
      "isReviewFormOpen": false,
      "isOwnReviewRequestFinished": false
    },
    "upNext": {
      "byId": {},
      "status": "IDLE"
    },
    "guestbooks": {
      "allIds": [],
      "byId": {},
      "activeId": null
    },
    "previousEpisode": {
      "byId": {},
      "statusById": {}
    }
  },
```

### Verbose log

```shell
[debug] Command-line config: ['-vU', '-i', '--all-subs', '--write-sub', '--cookies', 'path/to/cookies', '--format', 'bestvideo[height>=1080]+bestaudio/bestvideo+bestaudio/best', '--username', 'PRIVATE', '--password', 'PRIVATE', '-o', "Sabikui.Bisco.11.I'm Bisco\\!.WEBDL.mp4", 'https://beta.crunchyroll.com/watch/G4VUQZV4N/im-bisco']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.03.08.1 [c0c2c57d3] (zip)
[debug] Python version 3.8.10 (CPython 64bit) - Linux-5.4.0-104-generic-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 4.2.4, ffprobe N-103927-g68815d6791
[debug] Optional libraries: Crypto, secretstorage, sqlite
[debug] Proxy map: {}
Latest version: 2022.03.08.1, Current version: 2022.03.08.1
yt-dlp is up to date (2022.03.08.1)
[debug] [crunchyroll:beta] Extracting URL: https://beta.crunchyroll.com/watch/G4VUQZV4N/im-bisco
[crunchyroll:beta] im-bisco: Downloading webpage
ERROR: im-bisco: An extractor error has occurred. (caused by KeyError('G4VUQZV4N')); please report this issue on  https://github.com/yt-dlp/yt-dlp , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/local/bin/youtube-dl/yt_dlp/extractor/common.py", line 637, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/youtube-dl/yt_dlp/extractor/crunchyroll.py", line 719, in _real_extract
    episode_data = initial_state['content']['byId'][internal_id]
KeyError: 'G4VUQZV4N'
```

**Comment by krandor** [comment]
> I've reported a similar question/issue on CR beta. There's a draft fix mentioned in#2930 you can try out if you compile the changes yourself. Look at the proposed changes to readme in that draft to see new extractor args.

Thanks, I'll give that a try and report back!

**Comment by krandor** [comment]
I grabbed the commit and applied it to my local build and it's working! I'll keep this separate build for now, but I'll eagerly await for that PR to be completed.

#2930
